pyfar/dsp/regularization.py

```diff
@@ -0,0 +1,386 @@
+            ...
+            >>> pf.plot.freq(inverted, ax=axes[1], color='p',
+            ...     label='inverted with regulariztion')
+            >>> pf.plot.freq(1 / (sweep+1e-10), ax=axes[1], color='y',
+            ...     linestyle=':', label='inverted without regulariztion')
+            >>> axes[1].axvline(50, color='k', linestyle='--',
+            ...     label='frequency range')
+            >>> axes[1].axvline(16e3, color='k', linestyle='--')
+            >>> axes[1].set_ylim(-120, -20)
+            >>> axes[1].legend(loc='lower center')
+
+        Invert a headphone transfer function (HpTF), regularize the inversion at
+        high frequencies, and use a band-pass as target function. Note that the
+        equalized HpTF, which is obtained from a convolution of the HpTF with its
+        inverse filter, approaches the target function in both, time and frequency.
+
+        .. plot::
+
+            >>> import pyfar as pf
+            >>> import numpy as np
+            >>>
+            >>> # Hedphone transfer function to be inverted
+            >>> hptf = pf.signals.files.headphone_impulse_responses()[0, 0].flatten()
+            >>> # Regularize inversion at high frequencies
+            >>> regularization = pf.dsp.filter.low_shelf(
+            ...     pf.signals.impulse(hptf.n_samples), 4e3, -20, 2, 'II')
+            >>> # Use band-pass as target function
+            >>> target = pf.dsp.filter.butterworth(
+            ...     pf.signals.impulse(hptf.n_samples), 4, [100, 16e3], 'bandpass')
+            >>> target = pf.dsp.time_shift(target, 125, 'cyclic')
+            >>>
+            >>> # Regulated inversion
+            >>> Inversion = pf.dsp.RegularizedSpectrumInversion.from_magnitude_spectrum(
+            ...     hptf, regularization, beta=.1, target=target)
+            >>> inverted = Inversion.invert()
+            >>>
+            >>> # Plot the result
+            >>> ax = pf.plot.time_freq(hptf, label='HpTF')
+            >>> pf.plot.time_freq(inverted, label='HpTF inverted')
+            >>> pf.plot.time_freq(hptf * inverted, label='Equalized HpTF')
+            >>> pf.plot.time_freq(target, linestyle='--', label='Target')
+            >>> pf.plot.freq(regularization, ax=ax[1], label='Regularization')
+            >>> ax[0].set_xlim(0 , .005)
+            >>> ax[1].set_ylim(-40 , 15)
+            >>> ax[1].legend(loc='lower center', ncols=3)
+        """# noqa: E501
+        target = self.target
+        signal = self.signal
+
+        # zero pad, if target function is given and has different n_samples
+        if target is not None and \
+            target.n_samples != signal.n_samples:
+            n_samples = max(signal.n_samples, target.n_samples)
+            signal = pf.dsp.pad_zeros(signal,
+                                      (n_samples-signal.n_samples))
+            target = pf.dsp.pad_zeros(target,
+                                      (n_samples-target.n_samples))
```

Comment:
This might have been my idea - but I think we should undo this. If the target or signal is zero phase, padding zeros at the end would be a bad idea. I think it would be more save to force the user to take care of passing everything in the correct length.

Reply:
In this case using `target = self.target` and `signal = self.signal` might not be needed anymore.